### PR TITLE
[ROAD-322] Feat/add oss cache

### PR DESCRIPTION
**Pull request by aldanchenko**
Added cache for Open Source Security. If no changes in manifest files extension will return cached value. 

Screenshot 2022-01-26 at 15 22 45

### CODE REVIEW

File: Snyk.VisualStudio.Extension.Shared/Service/OssService.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            Logger.Information("Ignore Unknown CA is {IgnoreUnknownCA}", options.IgnoreUnknownCA);
+            Logger.Information("Additional Options is {AdditionalOptions}", options.AdditionalOptions);
+            Logger.Information("Is Scan All Projects is {IsScanAllProjects}", options.IsScanAllProjects);
+            Logger.Information("Solution path is {SolutionPath}", path);
+
+            Logger.Information("Start scan");
+
+            token.ThrowIfCancellationRequested();
+
+            var cliResult = this.cli.Scan(path);
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/Service/OssService.cs:56]: Aren't there any exceptions to be handled? (or to be sent to Sentry?)

File: Snyk.VisualStudio.Extension.Shared/Service/SnykTasksService.cs [hunk 2]
```diff
@@ -363,72 +358,34 @@ private void ScanOss(FeaturesSettings featuresSettings)
...
-
                             Logger.Information("Scan finished");
                         }
-                        catch (Exception e)
+                        catch (OssScanException e)
                         {
-                            bool isCancelled = this.cli.ConsoleRunner.IsStopped;
-
-                            this.cli = null;
+                            Logger.Error(e, "Oss scan exception");
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/Service/SnykTasksService.cs:381]: Is the logger is connected to Sentry via a Serilog-SInk? Just asking...

**Review by bastiandoetsch — APPROVED** [review]

File: Snyk.VisualStudio.Extension.Shared/Service/OssService.cs (hunk 1 above)
Comment (aldanchenko) [reply Snyk.VisualStudio.Extension.Shared/Service/OssService.cs:56]: Yes, this try/catch/finally exists on upper level - https://github.com/snyk/snyk-visual-studio-plugin/blob/102ca8fd97194e512c44780f90eb19254527d9e8/Snyk.VisualStudio.Extension.Shared/Service/SnykTasksService.cs#L371

File: Snyk.VisualStudio.Extension.Shared/Service/SnykTasksService.cs (hunk 2 above)
Comment (aldanchenko) [reply Snyk.VisualStudio.Extension.Shared/Service/SnykTasksService.cs:381]: Yes, Logger and Sentry configuration in one place here - https://github.com/snyk/snyk-visual-studio-plugin/blob/102ca8fd97194e512c44780f90eb19254527d9e8/Snyk.Common/LogManager.cs#L45

And LogManager used across all extension.

**Pull request merged**